src/clib/pioc_support.cpp

```diff
@@ -553,46 +553,115 @@ adios2_variable* spio_define_adios2_variable(iosystem_desc_t *ios, file_desc_t *
-        if (adiosErr != adios2_error_none)
+        /* Use lossless compression method by default */
+        int variable_compression_method = ios->adios_lossless_compression_method;
+
+        #ifdef _SPIO_ADIOS_USE_LOSSY_COMPRESSION
+        /* Switch to lossy compression method */
+        variable_compression_method = ios->adios_lossy_compression_method;
+
+        /* SZ or MGARD compressor only supports float or double types */
+        if (variable_compression_method == ADIOS_COMPRESSION_METHOD_SZ || variable_compression_method == ADIOS_COMPRESSION_METHOD_MGARD)
         {
-            pio_err(ios, file, PIO_EADIOS2ERR, __FILE__, __LINE__,
-                    "Failed to add Blosc2 compression operation (compressor=zstd) to variable %s (adios2_error=%s)",
-                    name, convert_adios2_error_to_string(adiosErr));
+            /* Switch back to lossless compression method for non floating point type */
+            if (type != adios2_type_float && type != adios2_type_double)
+                variable_compression_method = ios->adios_lossless_compression_method;
         }
+        #endif
 
-        /* Compression level is an integer between 0 (no compression) and 9 (more compression, more memory consumption) inclusive */
-        adiosErr = adios2_set_operation_parameter(variable, operation_index, "clevel", "5");
-        if (adiosErr != adios2_error_none)
+        if (variable_compression_method == ADIOS_COMPRESSION_METHOD_BLOSC2)
         {
-            pio_err(ios, file, PIO_EADIOS2ERR, __FILE__, __LINE__,
-                    "Failed to set Blosc2 compression level (clevel=5) for variable %s (adios2_error=%s)",
-                    name, convert_adios2_error_to_string(adiosErr));
-        }
+            if (ios->lossless_compression_operator != NULL)
+            {
+                /* Add a compression operation (Blosc2) to this variable */
+                adiosErr = adios2_add_operation(&operation_index, variable, ios->lossless_compression_operator, "compressor", "zstd"); /* Compressor type (e.g., zstd) */
+                if (adiosErr != adios2_error_none)
+                {
+                    pio_err(ios, file, PIO_EADIOS2ERR, __FILE__, __LINE__,
+                            "Failed to add Blosc2 compression operation (compressor=zstd) to variable %s (adios2_error=%s)",
+                            name, convert_adios2_error_to_string(adiosErr));
+                }
 
-        /* Shuffle option (BLOSC_SHUFFLE/BLOSC_NOSHUFFLE/BLOSC_BITSHUFFLE) */
-        adiosErr = adios2_set_operation_parameter(variable, operation_index, "doshuffle", "BLOSC_BITSHUFFLE");
-        if (adiosErr != adios2_error_none)
+                /* Compression level is an integer between 0 (no compression) and 9 (more compression, more memory consumption) inclusive */
+                adiosErr = adios2_set_operation_parameter(variable, operation_index, "clevel", "1");
```

Comment:
We also need to mention the default compression level used during configure. Eventually this should be configurable by the user.